src/common/transformations/src/transformations/common_optimizations/sdpa_scale_fusion.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/sdpa_scale_fusion.hpp"
+
+#include "openvino/core/rt_info.hpp"
+#include "openvino/core/type.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/scaled_dot_product_attention.hpp"
+#include "openvino/pass/pattern/op/pattern.hpp"
+#include "transformations/utils/gen_pattern.hpp"
+
+namespace ov {
+namespace pass {
+
+SDPAScaleFusion::SDPAScaleFusion() {
+    using namespace ov::pass::pattern;
+    using namespace ov::gen_pattern;
+
+    auto q = makePattern(ov::Rank(4));
+    auto k = makePattern(ov::Rank(4));
+    auto v = makePattern(ov::Rank(4));
+    auto mask = makePattern();
+    auto sdpa_scale = makePattern("[]");
+    auto scale_q = makePattern("[]");
+    auto scale_k = makePattern("[]");
+
+    auto scaled_q = makePattern<ov::op::v1::Multiply>({q, scale_q});
+    auto scaled_k = makePattern<ov::op::v1::Multiply>({k, scale_k});
```

Comment:
Doesn't it make sense to make one of these multiply optional?

Reply:
Relaxed pattern restrictions to support optional scales + non-constant scale case